### PR TITLE
Merge Taipy Core

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

**Review by trgiangdo — COMMENTED** [review]
The `src/taipy/core/tests` folder should be moved to the `src/tests/core` folder so that the tests action can run those.

**Review by jrobinAV — DISMISSED** [review]
Only 379 files to review...

File: src/__init__.py
Comment (jrobinAV) [inline src/__init__.py:1]: I believe this file should be deleted, no?.

**Comment by joaoandre-avaiga** [comment]
> The `src/taipy/core/tests` folder should be moved to the `src/tests/core` folder so that the tests action can run those.

I did move it but apparently I forgot to push 😅

**Review by jrobinAV — DISMISSED** [review]

**Comment by joaoandre-avaiga** [comment]
@jrobinAV There are some mypy issues that weren't previously checked in the Taipy repository. Most of them are on gui_core/ folder. I've solved some of them but decided to rollback because it would be kind of impossible to review on this PR.

I've temporarily disabled the mypy linter on this PR, but it should be back to normal on another PR. Also we should created a ticket to fix these issues later.

**Review by jrobinAV — COMMENTED** [review]
`src/taipy/rest/tests` folder and a lot of subfolders and subfiles have been removed while `rest/tests` have been created with only 2 files.

This looks strange to me.

**Comment by joaoandre-avaiga** [comment]
> `src/taipy/rest/tests` folder and a lot of subfolders and subfiles have been removed while `rest/tests` have been created with only 2 files.
> 
> This looks strange to me.

I noticed that the tests weren't move correctly on the rest pr. So on this one I've moved the tests inside src/taipy/rest/tests to tests/rest to fix it.

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**